### PR TITLE
feat: resources, attachments & materials

### PR DESCRIPTION
**Pull request by HuanCheng65**

### CODE REVIEW

File: src/materials/materials.service.ts [hunk 1]
```diff
@@ -78,69 +85,99 @@
...
+
+  async getMeta(
+    type: string,
     file: Express.Multer.File,
-  ): Promise<number> {
-    let meta;
+  ): Promise<PrismaJson.metaType> {
+    let meta: PrismaJson.metaType;
+
+    const buf = readFileSync(file.path);
```
[the remark this answers is not in the document]
Comment (nameisyui) [reply src/materials/materials.service.ts:95]: how to solve

**Comment by nameisyui** [comment]
@coderabbitai  In src/materials/materials.service.ts,there is an Uncontrolled data used in path expression.how to solve

File: src/materialbundles/materialbundles.service.ts [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+    if (!query) {
+      return {};
+    }
+
+    const conditions: Prisma.MaterialBundleWhereInput[] = [];
+    const regex = /(\w+)(:>=|:<=|:>|:<|:)(\w+)/g;
+    let match;
+
+    // Parsing special conditions like rating:>=4
+    while ((match = regex.exec(query)) !== null) {
```
[the remark this answers is not in the document]
Comment (nameisyui) [reply src/materialbundles/materialbundles.service.ts:144]: how can i fix this

[the remark this answers is not in the document]
Comment (nameisyui) [reply src/materialbundles/materialbundles.service.ts:144]: @coderabbitai review

**Comment by nameisyui** [comment]
@coderabbitai review

File: src/attachments/attachments.module.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import { Module } from '@nestjs/common';
+import { AttachmentsService } from './attachments.service';
+import { AttachmentsController } from './attachments.controller';
+import { MulterModule } from '@nestjs/platform-express';
+import { diskStorage } from 'multer';
+import { extname, join } from 'path';
```
[the remark this answers is not in the document]
Comment (nameisyui) [reply src/attachments/attachments.module.ts:6]: why ?

File: src/materials/materials.service.ts [hunk 4]
```diff
@@ -1,14 +1,21 @@
 import { Injectable, OnModuleInit } from '@nestjs/common';
-import { Material, MaterialType } from '@prisma/client';
+import { MaterialType } from '@prisma/client';
 import ffmpeg from 'fluent-ffmpeg';
 import path, { join } from 'path';
```
[the remark this answers is not in the document]
Comment (nameisyui) [reply src/materials/materials.service.ts:4]: why?

File: src/materialbundles/materialbundles.service.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (nameisyui) [reply src/materialbundles/materialbundles.service.ts:144]: do you have some suggestions?

File: prisma/schema.prisma [hunk 5]
```diff
@@ -418,15 +473,32 @@
...
 
+model MaterialbundlesRelation {
+  material   Material       @relation(fields: [materialId], references: [id])
+  materialId Int            @map("material_id")
+  bundle     MaterialBundle @relation(fields: [bundleId], references: [id], onDelete: Cascade)
+  bundleId   Int            @map("bundle_id")
+
+  @@id([materialId, bundleId])
+  @@map("materialbundles_relation")
+}
```
[the remark this answers is not in the document]
Comment (nameisyui) [reply prisma/schema.prisma:500]: do you have some suggestions?

File: src/attachments/attachments.controller.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    private readonly attachmentsService: AttachmentsService,
+    private readonly authService: AuthService,
+  ) {}
+
+  @Post()
+  @UseInterceptors(FileInterceptor('file'))
+  async uploadAttachment(
+    @Body() { type }: attachmentTypeDto,
+    @UploadedFile() file: Express.Multer.File,
+    @Headers('Authorization') auth: string | undefined,
```
[the remark this answers is not in the document]
Comment (nameisyui) [reply src/attachments/attachments.controller.ts:36]: ??why

File: prisma/schema.prisma [hunk 7]
```diff
@@ -12,6 +12,13 @@ generator client {
   binaryTargets = ["native", "debian-openssl-3.0.x", "debian-openssl-1.1.x"]
 }
 
+generator json {
+  provider  = "prisma-json-types-generator"
+  namespace = "PrismaJson"
+  // clientOutput = "<finds it automatically>"
+  // (./ -> relative to schema, or an importable path to require() it)
+}
```
[the remark this answers is not in the document]
Comment (nameisyui) [reply prisma/schema.prisma:20]: help me to add comments to explain the purpose and usage of the new JSON generator

**Comment by nameisyui** [comment]
@coderabbitai review

**Review by Nictheboy — APPROVED** [review]

**Pull request merged**